plugin/index.js

```diff
@@ -245,6 +245,12 @@ class ServerlessSecrets {
       if (!functions[functionName].environment) functions[functionName].environment = {}
       Object.assign(functions[functionName].environment, this.config.environments.$global, this.config.environments[functionName])
     })
+
+    // process.env.IS_LOCAL === 'true' is set when called by 'sls invoke local'
+    if (process.env._HANDLER === undefined && process.env.IS_LOCAL === 'true') {
+      const invokedFunction = functions[this.serverless.processedInput.options.function]
```

Comment:
Isn't `this.options` already the processed option object given to the plugin? If that's the case using this.options would decouple the implementation from Serverless internals.

Reply:
Yeap.

But there is a little difference.
If I call `sls invoke local --function some-function`:
```js
this.options {
  "function": "some-function"
}
this.serverless.processedInput.options {
  "commands": [
    "invoke",
    "local"
  ],
  "options": {
    "function": "some-function"
  }
}
```

And If I call `sls invoke local -f some-function`:
```js
this.options {
  "f": "some-function"
}
this.serverless.processedInput.options {
  "commands": [
    "invoke",
    "local"
  ],
  "options": {
    "f": "save-course",
    "function": "some-function"
  }
}
```

So using `this.serverless.processedInput.options` we always have `function` key.

In any case I could also do something like this:
```js
const invokedFunction = functions[this.options.f || this.options.function]
```

What do you think?

Reply:
Hmmm... That's strange also `serverless-webpack` uses `this.options.function` throughout the whole implementation. I was quite sure that the pluginManager would populate the long version in any case (regardless if you specify -f or --function). At least that was the handling for a long time.
However, the `const invokedFunction = ...` looks good otherwise.

Reply:
@HyperBrain 
You're right. I was calling  `JSON.stringify` inside the plugin constructor.

But if I call it later then the results are:
```js
"options": {
    "f": "save-course",
    "function": "save-course"
  }
```

It looks like `serverless` normalize it somehow, but I didn't take a deep look why this is happening.

So it's safe to use `this.options.function` as long as you are not using this in the plugin constructor.

Reply:
The reason why it is not yet normalized in the constructor is, that Serverless first loads the plugins and then starts the lifecycles of the invoked command. One of Serverless' lifecycle steps is the normalization (most likely in validate), but that can only be run as soon as Serverless knows the function (and thus the supported options, declared in the command structure).
So the same is true for any plugin - resolved and initialized properties (like options) are only available within the hooks, but not the constructor.

Reply:
Oh, I see.

Thanks for the explanation @HyperBrain